internal/gps/pkgtree/digest.go

```diff
@@ -150,110 +59,94 @@ type dirWalkClosure struct {
-		// declare type of the file system node.
-		modeType := info.Mode() & os.ModeType
-		var shouldSkip bool // skip some types of file system nodes
-
-		switch {
-		case modeType&os.ModeDir > 0:
-			mt = os.ModeDir
-			// DirWalkFunc itself does not need to enumerate children, because
-			// DirWalk will do that for us.
-			shouldSkip = true
-		case modeType&os.ModeSymlink > 0:
-			mt = os.ModeSymlink
-		case modeType&os.ModeNamedPipe > 0:
-			mt = os.ModeNamedPipe
-			shouldSkip = true
-		case modeType&os.ModeSocket > 0:
-			mt = os.ModeSocket
-			shouldSkip = true
-		case modeType&os.ModeDevice > 0:
-			mt = os.ModeDevice
-			shouldSkip = true
-		}
+			if len(osPathname) > closure.someDirLen {
+				osRelative = osPathname[closure.someDirLen:]
+			}
 
-		// Write the relative pathname to hash because the hash is a function of
-		// the node names, node types, and node contents. Added benefit is that
-		// empty directories, named pipes, sockets, devices, and symbolic links
-		// will also affect final hash value. Use `filepath.ToSlash` to ensure
-		// relative pathname is os-agnostic.
-		writeBytesWithNull(closure.someHash, []byte(filepath.ToSlash(osRelative)))
+			// Write the relative pathname to hash because the hash is a function of
+			// the node names, node types, and node contents. Added benefit is that
+			// empty directories, named pipes, sockets, devices, and symbolic links
+			// will also affect final hash value. Use `filepath.ToSlash` to ensure
+			// relative pathname is os-agnostic.
+			writeBytesWithNull(closure.someHash, []byte(filepath.ToSlash(osRelative)))
 
-		binary.LittleEndian.PutUint32(closure.someModeBytes, uint32(mt)) // encode the type of mode
-		writeBytesWithNull(closure.someHash, closure.someModeBytes)      // and write to hash
+			modeType := de.ModeType()
+			binary.LittleEndian.PutUint32(closure.someModeBytes, uint32(modeType)) // encode the type of mode
+			writeBytesWithNull(closure.someHash, closure.someModeBytes)            // and write to hash
 
-		if shouldSkip {
-			return nil // nothing more to do for some of the node types
-		}
+			switch {
+			case modeType&os.ModeSymlink != 0:
+				osRelative, err = os.Readlink(osPathname) // read the symlink referent
+				if err != nil {
+					return errors.Wrap(err, "cannot Readlink")
+				}
+				writeBytesWithNull(closure.someHash, []byte(filepath.ToSlash(osRelative))) // write referent to hash
+				return nil                                                                 // proceed to next node
+			case modeType&os.ModeDir != 0:
+				return nil // nothing more to do for this type
+			case modeType&os.ModeDevice != 0:
+				return nil // nothing more to do for this type
+			case modeType&os.ModeCharDevice != 0:
+				return nil // nothing more to do for this type
+			case modeType&os.ModeNamedPipe != 0:
+				return nil // nothing more to do for this type
+			case modeType&os.ModeSocket != 0:
+				return nil // nothing more to do for this type
```

Comment:
Could this block of cases be simplified a bit to something like this:
```go
case modeType&(os.ModeDir|os.ModeDevice|os.ModeCharDevice|os.ModeNamedPipe|os.ModeSocket):
  return nil // nothing more to do for these types
```

Reply:
Close:

```Go
case modeType&(os.ModeDir|os.ModeDevice|os.ModeCharDevice|os.ModeNamedPipe|os.ModeSocket) != 0:
```

Pushing update, and thanks!